tests/unit-tests/python/foglamp_test/plugins/south/test_http_south_device_unit.py

```diff
@@ -0,0 +1,111 @@
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("device")
+class TestHttpSouthDeviceUnit(object):
+    """Unit tests for foglamp.device.coap.IngestReadings
+    """
+
+    async def test_post_sensor_reading_ok(self):
+        data =  """{
+            "timestamp": "2017-01-02T01:02:03.23232Z-05:00",
+            "asset": "sensor1",
+            "key": "80a43623-ebe5-40d6-8d80-3f892da9b3b4",
+            "readings": {
+                "velocity": "500",
+                "temperature": {
+                    "value": "32",
+                    "unit": "kelvin"
+                }
+            }
+        }"""
+        with patch.object(Ingest, 'add_readings', return_value=asyncio.ensure_future(asyncio.sleep(0.1))) as mock_method1:
+            with patch.object(Ingest, 'is_available', return_value=True) as mock_method2:
+                request = mock_request(data)
+                r = await HttpSouthIngest.render_post(request)
+                retval = json.loads(r.body.decode())
+                # Assert the POST request response
+                assert 200 == retval['status']
+                assert 'success' == retval['result']
+
+
+    async def test_post_sensor_reading_missing_demiliter(self):
+        data =  """{
+            "timestamp": "2017-01-02T01:02:03.23232Z-05:00",
+            "asset": "sensor1",
+            "key": "80a43623-ebe5-40d6-8d80-3f892da9b3b4",
+            "readings": {
+                "velocity": "500",
+                "temperature": {
+                    "value": "32",
+                    "unit": "kelvin"
+                }
+        }"""
+        with patch.object(Ingest, 'add_readings', return_value=asyncio.ensure_future(asyncio.sleep(0.1))) as mock_method1:
+            with patch.object(Ingest, 'is_available', return_value=True) as mock_method2:
+                request = mock_request(data)
+                r = await HttpSouthIngest.render_post(request)
+                retval = json.loads(r.body.decode())
+                # Assert the POST request response
+                assert 400 == retval['status']
+                assert retval['error'].startswith("Expecting ',' delimiter:")
+
+
+    async def test_post_sensor_reading_not_dict(self):
+        data =  """{
+            "timestamp": "2017-01-02T01:02:03.23232Z-05:00",
+            "asset": "sensor2",
+            "key": "80a43623-ebe5-40d6-8d80-3f892da9b3b4",
+            "readings": "500"
+        }"""
+        with patch.object(Ingest, 'add_readings', return_value=asyncio.ensure_future(asyncio.sleep(0.1))) as mock_method1:
+            with patch.object(Ingest, 'is_available', return_value=True) as mock_method2:
+                request = mock_request(data)
+                r = await HttpSouthIngest.render_post(request)
+                retval = json.loads(r.body.decode())
+                # Assert the POST request response
+                assert 400 == retval['status']
+                assert "readings must be a dictionary" == retval['error']
+
```

Comment:
Running these tests generate warnings:
```
Task was destroyed but it is pending!
task: <Task pending coro=<sleep() done, defined at /usr/lib/python3.5/asyncio/tasks.py:502> wait_for=<Future pending cb=[Task._wakeup()]>>
Task was destroyed but it is pending!
task: <Task pending coro=<sleep() running at /usr/lib/python3.5/asyncio/tasks.py:516> wait_for=<Future pending cb=[Task._wakeup()]>>
```

Can we get rid of these?

Reply:
Will be resolved in some future sprint.